### PR TITLE
feat(p2p): refactor p2p configuration options

### PR DESCRIPTION
**Pull request by saghul**
Also add a preferH264 option for P2P.

### CODE REVIEW

File: JitsiConference.js [hunk 1]
```diff
@@ -151,7 +151,8 @@ export default function JitsiConference(options) {
      */
     this.deferredStartP2PTask = null;
 
-    const delay = parseInt(options.config.backToP2PDelay, 10);
+    const delay
+        = parseInt(options.config.p2p && options.config.p2p.backToP2PDelay, 10);
```
Comment (paweldomas) [inline JitsiConference.js]: I wish there was a nicer way to access these properties, but I can't think of any which would be simple so let it be...

**Comment by saghul** [comment]
Will do! Can anyone merge when tests have run? Thanks!

**Comment by paweldomas** [comment]
I suppose that the tests may actually fail, because those options are used there...

**Comment by saghul** [comment]
Yep, let me fixup those....

**Comment by damencho** [comment]
We had already pushed p2p into stable. Why not make the code to work with both configurations?
We had always try not to brake previous deployments. If we are braking it we try to announce it on the mailinglist, but as we are moving away from it ... maybe we should add a news update on the main site :)

**Comment by saghul** [comment]
Pushed a new commit enabling P2P in absence of new settings, PTAL.

**Comment by saghul** [comment]
Jenkins, test this please.

**Comment by saghul** [comment]
Results are in! 💚 http://ci.jitsi.org/job/jitsi-meet-PR-test/2075/

**Pull request merged**